### PR TITLE
Make JceMasterKey Case Insensitive

### PR DESCRIPTION
**Pull request by johnwalker**
Algorithm names in the JCA spec are not case-sensitive. This makes
JceMasterKey algorithm names case insensitive.

### CODE REVIEW

File: src/test/java/com/amazonaws/encryptionsdk/jce/JceMasterKeyTest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                "blatently/unsupported/algorithm");
+    }
+
+    /**
+     * Calls JceMasterKey.getInstance with differently cased wrappingAlgorithm names.
+     * Passes if no Exception is thrown.
+     * Relies on passing an invalid algorithm name to result in an Exception.
+     */
+    @Test
+    public void testGetInstanceCaseInsensitive() {
```
Comment (mattsb42-aws) [inline src/test/java/com/amazonaws/encryptionsdk/jce/JceMasterKeyTest.java]: These should be separate tests.

File: src/main/java/com/amazonaws/encryptionsdk/jce/JceMasterKey.java [hunk 2]
```diff
@@ -72,8 +72,8 @@
      */
     public static JceMasterKey getInstance(final SecretKey key, final String provider, final String keyId,
             final String wrappingAlgorithm) {
-        switch (wrappingAlgorithm) {
-            case "AES/GCM/NoPadding":
+        switch (wrappingAlgorithm.toUpperCase()) {
+            case "AES/GCM/NOPADDING":
```
Comment (mattsb42-aws) [inline src/main/java/com/amazonaws/encryptionsdk/jce/JceMasterKey.java]: GitHub won't let me comment on lines that weren't changed in the PR, but we also need to apply this type of check on line 98 where we check the `RSA/ECB/` prefix.

Comment (johnwalker) [reply src/main/java/com/amazonaws/encryptionsdk/jce/JceMasterKey.java]: Added test case for the RSA/ECB prefix. This check is already applied

File: src/test/java/com/amazonaws/encryptionsdk/jce/JceMasterKeyTest.java (hunk 1 above)
Comment (johnwalker) [reply src/test/java/com/amazonaws/encryptionsdk/jce/JceMasterKeyTest.java]: Separated the test into two cases

File: src/test/java/com/amazonaws/encryptionsdk/jce/JceMasterKeyTest.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        jceGetInstanceAsymmetric("rsa/ec/unsupportedAlgorithm");
+    }
+
+    /**
+     * Calls JceMasterKey.getInstance with differently cased wrappingAlgorithm names.
+     * Passes if no Exception is thrown.
+     * Relies on passing an invalid algorithm name to result in an Exception.
+     */
+    @Test
+    public void testGetInstanceCase1() {
```
Comment (mattsb42-aws) [inline src/test/java/com/amazonaws/encryptionsdk/jce/JceMasterKeyTest.java]: NP: Bytes are cheap. Human cognition trying to determine what a test is doing from a vague name is not.

Comment (johnwalker) [reply src/test/java/com/amazonaws/encryptionsdk/jce/JceMasterKeyTest.java]: Added more descriptive names.

**Pull request merged**